internal/services/housekeeper/tablePartitionCleaner.go

```diff
@@ -17,13 +17,13 @@ func DeletePartitions() {
 	retentionThresholdDate := currentTime.AddDate(0, 0, -cfg.DataRetentionPeriod)
 
 	// If the day of the month in $retentionThresholdDate is less than 15,
-	// set $partitionTableDate to the 16th of the previous month.
-	// Otherwise, set $partitionTableDate to the 1st of the current month.
+	// set $partitionTableDate to the 1st of the month.
+	// Otherwise, set $partitionTableDate to the 16th of the perivous month.
 	var partitionTableDate string
 	if retentionThresholdDate.Day() < 15 {
-		partitionTableDate = time.Date(retentionThresholdDate.Year(), retentionThresholdDate.Month()-1, 16, 0, 0, 0, 0, retentionThresholdDate.Location()).Format("2006-01-02")
-	} else {
 		partitionTableDate = retentionThresholdDate.AddDate(0, 0, -retentionThresholdDate.Day()+1).Format("2006-01-02")
```

Comment:
It should get the first day of the current month (as per the comment), right?
But for today's date, `partitionTableDate` is `2024-03-31` is it correct?

Reply:
My bad! I mistakenly checked the previous code. Please ignore this comment!

Reply:
So if we consider that job is going to run today then below should be the case.
currentTime = 2024-04-29
retentionThresholdDate = 2024-04-14
partitionTableDate = 2024-04-1 (so that all the partition before this date is removed)